modules/bar/battery/index.ts

```diff
@@ -59,29 +59,41 @@ const BatteryLabel = (): BarBoxChild => {
             ),
             visible: battery.bind('available'),
             tooltip_text: battery.bind('time_remaining').as((t) => t.toString()),
-            children: Utils.merge([battery.bind('available'), show_label.bind('value')], (batAvail, showLabel) => {
-                if (batAvail && showLabel) {
-                    return [
-                        Widget.Icon({
-                            class_name: 'bar-button-icon battery',
-                            icon: batIcon,
-                        }),
-                        Widget.Label({
-                            class_name: 'bar-button-label battery',
-                            label: battery.bind('percent').as((p) => `${Math.floor(p)}%`),
-                        }),
-                    ];
-                } else if (batAvail && !showLabel) {
-                    return [
-                        Widget.Icon({
-                            class_name: 'bar-button-icon battery',
-                            icon: batIcon,
-                        }),
-                    ];
-                } else {
-                    return [];
-                }
-            }),
+            children: Utils.merge(
+                [
+                    battery.bind('available'),
+                    show_label.bind('value'),
+                    battery.bind('charged'),
+                    hideLabelWhenFull.bind('value'),
+                ],
+                (batAvail, showLabel, hideWhenFull, isCharged) => {
```

Comment:
```suggestion
                    hideLabelWhenFull.bind('value'),
                ],
                (batAvail, showLabel, isCharged, hideWhenFull) => {
```
These two are swapped.